brainscore/benchmarks/search.py

```diff
@@ -85,3 +85,84 @@ def ceiling(self):
 
     def get_raw_data(self):
         return self.cumm_perf, self.saccades
+
+KlabZhang2018search_assembly = ['klab.Zhang2018search_waldo', 'klab.Zhang2018search_naturaldesign']
+KlabZhang2018search_assembly_ceil = [1, 1]
+
+class KlabZhang2018VisualSearch(Benchmark):
+    def __init__(self, ceil_score=None, assembly_name=None):
+        if assembly_name is None:
+            self._logger.info("## No assembly name given, therefore selecting random assembly from 'waldo' or 'naturaldesign'...\n")
+            i = np.random.choice(2)
+            assembly_name = KlabZhang2018search_assembly[i]
+            ceil_score = KlabZhang2018search_assembly_ceil[i]
+
+        self.human_score = Score([ceil_score, np.nan],
+                        coords={'aggregation': ['center', 'error']}, dims=['aggregation'])
+        self._version = 1
+        self._identifier='klab.Zhang2018-visual_search'
+        self.parent = 'visual_search'
+        self.paper_link='https://doi.org/10.1038/s41467-018-06217-x'
+        self._assemblies = brainscore.get_assembly(assembly_name)
+        self._stimuli = self._assemblies.stimulus_set
+
+        self.max_fix = self._assemblies.fixation.values.shape[0] - 2
+        self.num_sub = np.max(self._assemblies.subjects.values)
+        self.data_len = int(self._assemblies.presentation.values.shape[0]/self.num_sub)
+        self.ior_size = 100
+
+        self._logger = logging.getLogger(fullname(self))
+
+    def __call__(self, candidate: BrainModel):
+        self._metric = ScanMatchPy.initialize()
+
+        self._logger.info("## Starting visual search task...")
+        candidate.start_task(BrainModel.Task.visual_search, max_fix=self.max_fix, data_len=self.data_len, ior_size=self.ior_size)

```

Comment:
the `data_len` is still a little unclear to me -- why does the model need to know about this?

Reply:
I will make the changes we can use stimulus_set for that. I added this parameter inside the benchmark init. So basically, we need data_len at some other places also. so instead of computing every time from the stimulus_set, I thought it may be much better to assign a variable to it. Note that the length of stimulus_set will be 3 times the data_len.